tests/parser-test-files/lisp-toplevel.txt

```diff
@@ -0,0 +1,22 @@
+================================================================================
```

Comment:
can you add a test that just has plain invalid Lisp syntax inside of a lisp-toplevel form?

Reply:
Invalid lisp syntax like `(defmacro 777)` will only be detected at parse time if an eval-when clause is detected and contents are evaluated. I can a handler and tests for this case.

Hard syntax errors like unterminated strings, unknown character literals, and so on will signal inside of eclector, before anything in this changeset is run. For instance, an unterminated `(package` form produces:

```
Unhandled ECLECTOR.READER:UNTERMINATED-LIST in thread #<SB-THREAD:THREAD tid=259 "main thread" RUNNING
                                                         {7004FD0733}>:
  While reading list, expected the character ) when input ended.

Backtrace for: #<SB-THREAD:THREAD tid=259 "main thread" RUNNING {7004FD0733}>
```
This is existing (not so good) behavior -- leaving unterminated forms in, e.g. library/types.lisp causes: 
```
Unhandled ECLECTOR.READER:UNTERMINATED-LIST in thread #<SB-THREAD:THREAD tid=259 "main thread" RUNNING
                                                         {7004FD0733}>:
  While reading list, expected the character ) when input ended.

Backtrace for: #<SB-THREAD:THREAD tid=259 "main thread" RUNNING {7004FD0733}>
```

The third case is lisp code that isn't evaluated at parse time, and is emitted unchanged to the .lisp output that is subsequently given to compile-file. I'm not sure how to handle that one, though, I am pretty sure that it would not be hard to map an error offset thrown from compile-file back to an originating form.

Reply:
file a ticket for the 2nd case